### DISCUSSION TITLE
im getting the 504 error

### DISCUSSION THREAD

**Discussion by rohitdr** [vercel/community#2011] [Help] [answered]
### Summary

i m getting the 504 error

**Comment by amyegan** [comment, marked as answer]
Hi @rohitdr! A `504` [FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#function_invocation_timeout) or [EDGE_FUNCTION_INVOCATION_TIMEOUT](https://vercel.com/docs/errors/application-error-codes#edge_function_invocation_timeout) error happens when a function takes too long. Common reasons for this are because of a delay when calling an external service, like a database or API, or because data processing took too long in the function. 

The time limit for Edge Function execution is 25 seconds before timeout. Serverless Function timeout happens at 10 seconds for Hobby accounts and 15 seconds for Pro accounts. However, Pro customers can raise the [max duration](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) up to 5 minutes. You can learn more about these limits in our [Limits & Limitations](https://vercel.com/docs/concepts/limits/overview#) doc.

**Reply by adobayua-rpay** [reply]
Hi Amy, wondering why this may not be resolved using services like Azure functions? Our calls take 5 minutes+, we moved them over to TimerTriggers on Azure but we still get 504 responses either way.
Any help would be appreciated, thanks!

**Reply by vpaflah** [reply]
I got the same issue with a simple hello world api.

```
export default function handler(req, res) {
  res.status(200).json({ message: 'Hello from Glomium, Server is running properly!' })
}
```

you can check the api from [here](https://glomium-dashboard.vercel.app/api/hello/)

Do you have any solution?

**Reply by FitiavanaLancelot7** [reply]
Same as well even I'm using an asynchronious function

**Reply by amyegan** [reply]
Hi! This is a fairly old post. If you're having a new issue with function timeouts, please first try the [troubleshooting tips for 504 errors](https://vercel.com/docs/errors). If you are still stuck after that, please [start a new discussion](https://github.com/vercel/community/discussions/new) with a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) and explanation of the steps we should follow to see the error ourselves. If you have [runtime log](https://vercel.com/docs/observability/runtime-logs) info available to share, that would also be helpful.
